deepchem/data/data_loader.py

```diff
@@ -437,6 +439,124 @@ def _featurize_shard(self, shard):
+  pandas, but this class may prove useful if you're processing
+  large json files that you don't want to manipulate directly in
+  memory.
+
+  It is meant to load JSON files formatted as "records" in line
+  delimited format, which allows for sharding.
+  ``list like [{column -> value}, ... , {column -> value}]``.
+
+  Examples
+  --------
+  >> import pandas as pd
+  >> df = pd.DataFrame(some_data)
+  >> df.columns.tolist()
+  .. ['formula', structure', 'task']
+  >> df.to_json('file.json', orient='records', lines=True)
+  >> loader = JsonLoader(['task'], {'structure': dict}, 'formula')
+  >> dataset = loader.create_dataset('file.json')
+  
+  """
+
+  def __init__(self,
+               tasks: List[str],
+               json_fields: Dict[str, type],
+               id_field: str = None,
+               featurizer: Optional[Featurizer] = None,
+               log_every_n: int = 1000):
+    """Initializes JsonLoader.
+
+    Parameters
+    ----------
+    tasks : List[str]
+      List of task names
+    json_fields : Dict[str, type]
+      column names and dtypes in dataframe containing data to be featurized
+      e.g. {"structure": dict, "composition": str}
+    id_field : str, default None
+      Column for identifying samples.
+    featurizer : dc.feat.Featurizer, optional
+      Featurizer to use to process data
+    log_every_n : int, optional
+      Writes a logging statement this often.
+
+    """
+
+    if not isinstance(tasks, list):
+      raise ValueError("Tasks must be a list.")
+    self.tasks = tasks
+    self.json_fields = json_fields
+    if id_field is None:
+      self.id_field = next(iter(json_fields))
+    else:
+      self.id_field = id_field
+
+    self.user_specified_features = None
+    if isinstance(featurizer, UserDefinedFeaturizer):
+      self.user_specified_features = featurizer.feature_fields
+    self.featurizer = featurizer
+    self.log_every_n = log_every_n
+
+  def _get_shards(self, input_files, shard_size):
+    """Defines a generator which returns data for each shard"""
+    return load_json_files(input_files, shard_size)
+
+  def _featurize_shard(self, shard):
+    """Featurizes a shard of an input dataframe."""
+    return self._featurize_df(
+        shard,
+        self.featurizer,
+        json_fields=self.json_fields,
+        log_every_n=self.log_every_n)
+
+  def _featurize_df(self,
+                    shard,
+                    featurizer: Featurizer,
+                    json_fields: Dict[str, type],
+                    log_every_n: int = 1000):
+    """Featurize individual materials in dataframe.
+
+    Helper that given a featurizer that operates on individual
+    inorganic crystal structures, computes & adds features for
+    that compound to the features dataframe.
+
+    Parameters
+    ----------
+    shard: pd.DataFrame
+      DataFrame that holds pymatgen.Structure dict or 
+      pymatgen.Composition str
+    featurizer: CrystalFeaturizer
+      A crystal featurizer object
```

Comment:
This documentation should be more general.  It's not just for inorganic crystals, and it allows any kind of data with any featurizer.

Reply:
Sounds good, I'll fix this.